core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -685,6 +675,8 @@ impl<S: ReadStorage> VmInterfaceHistoryEnabled for Vm<S> {
             self.snapshot.is_none(),
             "cannot create a VM snapshot until a previous snapshot is rolled back to or popped"
         );
+
+        self.delete_history_if_appropriate();
```

Comment:
@joonazan Is this additional call acceptable? It removes unnecessary bootloader heap history recorded outside of a snapshot, so that VM state assertions work as expected. I suppose another way to approach this issue would be to change state comparison logic, but I think deleting history makes sense semantically.

Reply:
I see, RollbackableMap and heap history accumulate even when there is no snapshot. That is somewhat undesirable, so I think this is a positive change.

Reply:
If the VM stored the external snapshot, it would be able to not build up history when there is no snapshot. This was a significant improvement for API speed in the old VM.

On the other hand, if we want to go for the ability to rewind the VM to any historical point in time, we'd want to never clear history.